cpu/stm32/periph/gpio_f1.c

```diff
@@ -241,12 +243,16 @@ void gpio_irq_disable(gpio_t pin)
 void isr_exti(void)
 {
     /* only generate interrupts against lines which have their IMR set */
-    uint32_t pending_isr = (EXTI->PR & EXTI->IMR);
-    for (unsigned i = 0; i < GPIO_ISR_CHAN_NUMOF; i++) {
-        if (pending_isr & (1 << i)) {
-            EXTI->PR = (1 << i);        /* clear by writing a 1 */
-            exti_ctx[i].cb(exti_ctx[i].arg);
-        }
+    uint32_t pending_isr = (EXTI->PR & EXTI->IMR & GPIO_ISR_CHAN_MASK);
```

Comment:
Is the `GPIO_IRQ_CHAN_MASK` actually needed, or is this just better safe than sorry?

Reply:
Aha, yes! https://github.com/RIOT-OS/RIOT/pull/15899

Reply:
But let's get https://github.com/RIOT-OS/RIOT/pull/15899 in first, so that it is clear from the commit history what is the bug fix and what is the optimization.

Reply:
#15899 has been merged ;)